providers/src/airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -323,6 +333,9 @@ def get_sqlalchemy_engine(self, engine_kwargs=None):
         if "insecure_mode" in conn_params:
             engine_kwargs.setdefault("connect_args", {})
             engine_kwargs["connect_args"]["insecure_mode"] = True
+        if "json_result_force_utf8_decoding" in conn_params:
```

Comment:
I thought about lumping this together into the same branch on line 333 above, but wanted to reduce diff count for now. Happy to make that change as well if you prefer, since it's the same type of value.